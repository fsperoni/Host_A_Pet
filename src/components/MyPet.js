import React, { useContext, useState } from "react";
import UserContext from "../hooks/useUserContext";
import HostAPetApi from "./Api";
import EditPetForm from "./EditPetForm";

const MyPet = ({ pet, pets, setPets }) => {
  const [showEditForm, setShowEditForm] = useState(false);
  const { currentUser } = useContext(UserContext);

  const handleEditPet = () => {
    setShowEditForm(!showEditForm);
  }

  const handleDeletePet = async(evt) => {
    const id = parseInt(evt.target.children[0].value);
    await HostAPetApi.deletePet(currentUser.username, id);
    setPets(pets => pets.filter(pet => pet.id !== id));
  }

  return (
  <div key={pet.id} className="card mt-3">
      <div className="card-body">
        <h3>{pet.name} the {pet.type}</h3>
        <img src={pet.photo} className="mt-2 img-fluid" alt="pet"/>
        <div className="container mt-3">
          <button className="btn btn-sm btn-primary" onClick={handleEditPet}>
            Edit
          </button>
          <button className="btn btn-sm btn-danger ms-2" onClick={handleDeletePet}>
            <data value={pet.id} />
            Delete
          </button>
          {showEditForm ? 
            <EditPetForm pet={pet} pets={pets} setPets={setPets} setShowEditForm={setShowEditForm}/> : 
          null}
        </div>
      </div>
    </div>
  )
}

export default MyPet;