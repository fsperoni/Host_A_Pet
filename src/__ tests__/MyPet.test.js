import { render } from "@testing-library/react";
import MyPet from "../components/MyPet";
import UserContext from "../hooks/useUserContext";

const {
  currentAdminUser,
  pet, pets
} = require("./_testCommon");

// smoke test
it('renders without crashing', function() {
  render(
    <UserContext.Provider
          value={{ currentUser: currentAdminUser() , function() {}}}>
      <MyPet pet={pet()} pets={pets()}/>
    </UserContext.Provider>);
});

//snapshot test
it("matches snapshot", function() {
  const {asFragment} = render(
    <UserContext.Provider
          value={{ currentUser: currentAdminUser() , function() {}}}>
      <MyPet pet={pet()} pets={pets()}/>
    </UserContext.Provider>
  );
  expect(asFragment()).toMatchSnapshot();
});

it("should have 'Garfield1 the Dog' in the component.", function() {
  const { getByText } = render(
    <UserContext.Provider
          value={{ currentUser: currentAdminUser() , function() {}}}>
      <MyPet pet={pet()} pets={pets()}/>
    </UserContext.Provider>
  );
  expect(getByText("Garfield1 the Dog")).toBeInTheDocument();
});
