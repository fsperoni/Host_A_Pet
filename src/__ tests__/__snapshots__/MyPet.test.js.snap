// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="card mt-3"
  >
    <div
      class="card-body"
    >
      <h3>
        Garfield1 the Dog
      </h3>
      <img
        alt="pet"
        class="mt-2 img-fluid"
        src="/static/media/dog.4b90f018c1cda3590605.jpeg"
      />
      <div
        class="container mt-3"
      >
        <button
          class="btn btn-sm btn-primary"
        >
          Edit
        </button>
        <button
          class="btn btn-sm btn-danger ms-2"
        >
          <data
            value="23"
          />
          Delete
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
